### ISSUE TITLE
[BUG] Refresh browser for the virtualization management page of accessing Harvester would display empty or page not found error

### ISSUE THREAD

**Issue by TachunLin** [rancher/dashboard#11350]
**Rancher Server Setup**
- Rancher version: `v2.9.0-alpha5` and `v2.9-head` (4c5e874)
- Installation option (Docker install/Helm Chart): Helm Chart

**Harvester Setup**
- Harvester version: `v1.3.1` and `v1.2.2`

**Describe the bug**

With Harvester have imported in Rancher
When access Harvester in Rancher virtualization management. 

* After the Harvester cluster content come up
    ![image](https://github.com/rancher/rancher/assets/29251855/9b1b11a0-5a67-41ed-bea0-b0e714161d3d)

* On Harvester **v1.3.1** + Rancher **v2.9.5-alpha5**
  - When we try to fresh the page via browser function (or F5), It would display empty page content 
  
    https://github.com/rancher/dashboard/assets/29251855/46dee9f4-3a7a-4578-9091-2134e90798a0

* On Harvester **v1.2.2** + Rancher **v2.9-head** 
  - When we try to fresh the page via browser function (or F5), It would display `Error 404: This page could not be found`
  
  ![image](https://github.com/rancher/dashboard/assets/29251855/15ec113f-b49b-4582-9605-d908d0095a9e)

**To Reproduce**

1. Use helm to provision the Rancher v2.9.0-alpha5 
    - Repo url: https://releases.rancher.com/server-charts/alpha
2. Create a v1.3.1 Harvester cluster
3. Import Harvester to Rancher
4. Access Harvester from Rancher virtualization management page
5. Refresh the page

**Expected Result**

Every time refresh the Harvester cluster content page should display all content well.

**Comment by richard-cox** [comment]
This is solved when running locally, but not in a production build.

In addition, when not logged in and navigating to a harvester cluster's page, there's a different error

```
getters.js:24 Uncaught (in promise) Error: Unknown schema for type: management.cattle.io.cluster
    at getters.js:24:13
```
which ultimately points at a findAll action. That part is reproducible in both local and production builds

**Comment by gaktive** [comment]
@TachunLin how's testing for this looking? 2.9.0's release date is targeted for next week and QA is wrapping up other tickets in the next day or so.

**Comment by IsaSih** [comment]
Tests pass on v2.9-dd23578d3e3499f54d80c4ed88cc52c44ee69539-head

**Comment by TachunLin** [comment]
Verified **fixed** on Rancher `v2.9-6e9cf233a87796ee9a96e73c096a82101fc9c2ff-head` with Harveter `v1.3.1` and `v1.2.2`.

### Result

1. On Rancher **v2.9-head** with Harvester **v1.3.1**, after refreshing the virtualization management page, the content can loaded well without empty. 


  https://github.com/user-attachments/assets/0b794592-bf0b-4592-b86d-106ea8d299ea

2. On Rancher **v2.9-head** with Harvester **v1.2.2**, after refreshing the virtualization management page, the content can loaded well without showing `Error 404: This page could not be found`


  https://github.com/user-attachments/assets/deea3d0e-1d1d-4c57-8b08-008cfb280ef9



### Test Information
* Test Environment: Single nodes harvester on kvm machine
* Harvester version: `v1.3.1` and `v1.2.2`
* Rancher version: `v2.9-6e9cf233a87796ee9a96e73c096a82101fc9c2ff-head`
